File: frontend/src/api/certificate.ts
import { get, post, del, getPage } from './request';
import type { Certificate, PageResponse } from '@/types';

export interface CertificateQueryParams {
  page?: number;
  pageSize?: number;
  search?: string;
  status?: string;
  sortBy?: string;
}

export interface CertificateOption {
  id: string;
  name: string;
  domain: string;
}

export interface ApplyCertificateData {
  caProvider: string;
  domain: string;
  csrId: number;
  credentialId: string;
  validateType: string;   // DNS / FILE
  productType: string;    // DV / OV / EV
  domainType: string;     // single / wildcard / multi
}

export interface ImportCertificateData {
  certPEM: string;
  chainPEM?: string;
  privateKeyPEM?: string;
}

// 获取证书列表（分页）
export function getCertificates(params: CertificateQueryParams): Promise<PageResponse<Certificate>> {
  return getPage<Certificate>('/certificates', params as Record<string, unknown>);
}

// 获取单个证书详情
export function getCertificate(id: string): Promise<Certificate> {
  return get<Certificate>(`/certificates/${id}`);
}

// 获取证书选项列表（简化列表，用于下拉选择）
export function getCertificateOptions(): Promise<CertificateOption[]> {
  return get<CertificateOption[]>('/certificates/options');
}

// 删除证书
export function deleteCertificate(id: string): Promise<void> {
  return del<void>(`/certificates/${id}`);
}

// 上传证书
export function uploadCertificate(data: FormData): Promise<Certificate> {
  return post<Certificate>('/certificates/upload', data, {
    headers: {
      'Content-Type': 'multipart/form-data',
    },
  });
}

// 导入证书
export function importCertificate(data: ImportCertificateData): Promise<Certificate> {
  return post<Certificate>('/certificates/import', data);
}

// 申请证书
export function applyCertificate(data: ApplyCertificateData): Promise<Certificate> {
  return post<Certificate>('/certificates/apply', data);
}

// 同步证书状态
export function syncCertStatus(id: string): Promise<Certificate> {
  return post<Certificate>(`/certificates/${id}/sync`);
}

File: backend/internal/model/certificate.go
package model

import (
	"time"
)

// Certificate 证书模型
type Certificate struct {
	BaseModel
	Domain              string    `gorm:"size:255;not null;index" json:"domain"`
	CAProvider          string    `gorm:"size:100" json:"ca_provider"`
	Status              string    `gorm:"size:50;not null" json:"status"`
	ExpireAt            time.Time `json:"expire_at"`
	CSRID               uint      `gorm:"index" json:"csr_id"`
	CredentialID        uint      `gorm:"index" json:"credential_id"` // 关联的云凭证 ID
	CertPEM             string    `gorm:"type:text" json:"-"`
	ChainPEM            string    `gorm:"type:text" json:"-"`
	PrivateKeyEncrypted string    `gorm:"type:text" json:"-"`
	Issuer              string    `gorm:"size:255" json:"issuer"`
	Fingerprint         string    `gorm:"size:255;index" json:"fingerprint"`
	KeyAlgorithm        string    `gorm:"size:50" json:"key_algorithm"`
	SerialNumber        string    `gorm:"size:255" json:"serial_number"`
	OrderID             string    `gorm:"size:255" json:"order_id"`     // CA返回的订单ID，用于后续状态查询
	VerifyType          string    `gorm:"size:20" json:"verify_type"`   // 验证方式：DNS / FILE
	VerifyInfo          string    `gorm:"type:text" json:"verify_info"` // JSON格式的验证信息
	ProductType         string    `gorm:"size:20" json:"product_type"`  // 证书级别：DV / OV / EV
	DomainType          string    `gorm:"size:20" json:"domain_type"`   // 域名类型：single / wildcard / multi
}

// TableName 指定表名
func (Certificate) TableName() string {
	return "certificates"
}
